File: contest/26/b/main.py
def min(a: int, b: int) -> int:
    return a if a < b else b


def rec(actual: int, rest: int) -> int:
    if rest == 4:
        return actual * 100 + 51
    if rest == 5:
        return rec(actual * 10 + 5, 4)
    if rest == 6:
        return rec(actual * 100 + 50, 4)
    return rec(actual * 100 + 50, rest - 2)


def solve():
    n = int(input())
    if n == 1:
        print(-1)
        return
    if n == 2:
        print(66)
        return
    if n == 3:
        print(-1)
        return

    print(66 * rec(0, n))


t = int(input())
for _ in range(0, t):
    solve()

File: contest/26/b/main.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = long long;
#define int ll
#define endl (char)10

void __dbg() { cerr << endl; }
template <typename T> void __dbg(T t) { cerr << t << endl; }
template <typename T, typename... TRest> void __dbg(T first, TRest... rest) {
  cerr << first << ", ";
  __dbg(rest...);
}
#define dbg(...)                                                               \
  do {                                                                         \
    cerr << "DBG> " << "(" << #__VA_ARGS__ << ") = ";                          \
    __dbg(__VA_ARGS__);                                                        \
  } while (0)

int rec(int actual, int rest) {
  if (rest == 0) {
    return actual % 66 == 0 ? actual : -1;
  }

  int first = rec(actual * 10 + 3, rest - 1);
  int second = rec(actual * 10 + 6, rest - 1);
  if (first != -1 && second != -1) {
    return min(first, second);
  }
  if (first == -1 && second == -1) {
    return -1;
  }
  if (first == -1) {
    return second;
  }

  return first;
}
void solution() { int n; }

int32_t main() {
  ios_base::sync_with_stdio(false);
  cin.tie(0), cout.tie(0);
  int t;
  cin >> t;
  while (t--) {
    solution();
  }

  return 0;
}
